core/src/env/arithmetic.rs

```diff
@@ -92,7 +96,7 @@ pub trait BaseArithmetic:
 
 impl<T> BaseArithmetic for T where
     T: Sized
-        + From<u32>
+        + From<u8>
```

Comment:
Can you explain why this change is required for this PR?

Reply:
Yes, this brings it into line with the substrate `BaseArithmetic`. The `From<u32>` bound is now defined in the `AtLeast32Bit` trait, and it is assumed for `BaseArtithmetic` trait that we are least dealing with `u8`

Reply:
> Can you explain why this change is required for this PR?

It's not strictly required for this PR, it's just something I overlooked in #463.

Reply:
Ah okay so am I right that the above changes actually are aligning ink! further with Substrate traits? Just want to be sure about this because I really am not familiar with the Substrate tests but getting traits righti is really really important.

Reply:
Yes exactly. 

We are still missing a few more e.g. the rest of the `Checked*` traits and `UniqueSaturatedFrom`/ `UniqueSaturatedInto`, but these are definitely out of scope for this PR.

```
impl<T: Sized, S: TryFrom<T> + Bounded + Sized> UniqueSaturatedFrom<T> for S {
	fn unique_saturated_from(t: T) -> Self {
		S::try_from(t).unwrap_or_else(|_| Bounded::max_value())
	}
}

impl<T: Bounded + Sized, S: TryInto<T> + Sized> UniqueSaturatedInto<T> for S {
	fn unique_saturated_into(self) -> T {
		self.try_into().unwrap_or_else(|_| Bounded::max_value())
	}
}
```